packages/react-tooltip/Spec.md

```diff
@@ -1,63 +1,515 @@
+ */
+export function useTooltipSlot(
+  state: React.HTMLAttributes<HTMLElement> & WithTooltipSlot,
+  options?: TooltipSlotOptions,
+);
+```
+
+### TooltipTrigger
+
+`TooltipTrigger` allows tooltips to be added to any focusable component that doesn't have its own `tooltip` prop. It supports a single JSX child, and works by cloning the JSX element in order to add the same `onPointerDown`, etc. listeners that are added by `useTooltipSlot`.
+
+```typescript
+export interface TooltipTriggerProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * The tooltip content to display on hover or focus. Can be a string, JSX element tree, or TooltipProps.
+   */
+  tooltip?: ShorthandProps<TooltipProps>;
+
+  /**
+   * The aria property that will be set to the tooltip's ID.
+   *
+   * @defaultvalue aria-describedby
+   */
+  ariaProp?: 'aria-describedby' | 'aria-labelledby';
+}
+
+export const tooltipTriggerShorthandProps = ['tooltip'] as const;
+```
+
+## @fluentui/react-tooltip
+
+The `react-tooltip` package contains the bulk of the implementation of tooltips, including rendering, styling, positioning, lifetime management, etc.
+
+### Tooltip
+
+`Tooltip` renders the tooltip content itself, and the arrow that points to the target element. The tooltip renders in a React portal to avoid clipping.
+
+The `TooltipProps` interface is defined in the `react-tooltip-hook` package.
+
+The `TooltipState` type has a slot for the arrow. This arrow slot can't be set by a prop, but is used internally for rendering and positioning.
+
+```typescript
+import { TooltipProps } from '@fluentui/react-tooltip-hook';
+
+export { TooltipProps };
+
+export const tooltipShorthandProps = [] as const;
+export const tooltipSlotProps = [...tooltipShorthandProps, 'arrow'] as const;
+
+export type TooltipState = ComponentState<
+  TooltipProps & {
+    /**
+     * The arrow that points to the target element.
+     */
+    arrow?: ObjectShorthandProps<React.HTMLAttributes<HTMLElement> & React.RefAttributes<HTMLElement>>;
+  },
+  /* ShorthandProps: */ typeof tooltipShorthandProps[number],
+  /* DefaultedProps: */ 'arrow' | 'placement'
+>;
+```
+
+### TooltipProvider
+
+`TooltipProvider` is responsible for providing the actual implementation of `TooltipManagerApi` and `Tooltip`. It uses a React context to that contains those when it is included in the tree. This context will also be built into `FluentContext`.
+
+```typescript
+export interface TooltipProviderProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  // TooltipProvider has no additional props
+}
+
+export const tooltipProviderShorthandProps = [] as const;
+
+export type TooltipProviderState = ComponentState<TooltipProviderProps, typeof tooltipProviderShorthandProps[number]>;
+```
+
+### TooltipManager
+
+`TooltipManager` handles showing and hiding tooltips, including positioning and tracking the visible tooltip. There is only one instance at the root of the app.
+
```

Comment:
This Codesandbox shows the scenario with multiple documents: https://codesandbox.io/embed/7hoso
(host application creates an iframe and then mounts a React portal into it)

I don't think that we should handle now "one tooltip per document", but portals should be mounted into a proper document. Please let me know if you will need any help with it 😉